src/dfx/src/lib/operations/ledger.rs

```diff
@@ -164,6 +168,36 @@ pub async fn transfer(
     Ok(block_height)
 }
 
+fn compose_insufficient_funds_error(
+    agent: &Agent,
+    subaccount: Option<Subaccount>,
+    balance: ICPTs,
+) -> DfxResult<BlockHeight> {
+    let principal = agent.get_principal().unwrap(); // This should always succeed at this point.
+
+    let explanation = "Insufficient ICP balance to finish the transfer transaction.";
+    let suggestion = format!(
+        "Please top up your ICP balance.
+
+Your account address for receiving ICP from centralized exchanges: {}
+(run `dfx ledger account-id` to display)
+
+Your principal for ICP wallets and decentralized exchanges:{}
```

Comment:
for consistency
```suggestion
Your principal for ICP wallets and decentralized exchanges: {}
```

Reply:
Fixed at https://github.com/dfinity/sdk/pull/4019/commits/0c9b12feb620103990f5033b8163447e3122748d